projects/packages/publicize/src/rest-endpoints/class-connections-controller.php

```diff
@@ -0,0 +1,302 @@
+<?php
+/**
+ * The Publicize Connections Controller class.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize\Rest_Endpoints;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager;
+use Automattic\Jetpack\Publicize\Connection_Fields;
+use WP_Error;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * Registers the REST routes
+ */
+class Connections_Controller extends WP_REST_Controller {
+
+	/**
+	 * Whether we are on WPCOM.
+	 *
+	 * @var bool $is_wpcom
+	 */
+	protected $is_wpcom = false;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v3';
+		$this->rest_base = 'publicize/connections';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+
+		$this->is_wpcom = defined( 'IS_WPCOM' ) && IS_WPCOM;
+
+		$this->wpcom_is_wpcom_only_endpoint = true;
+	}
+
+	/**
+	 * Register the routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permission_check' ),
+					'args'                => array(
+						'test_connections' => array(
+							'type'        => 'boolean',
+							'description' => __( 'Whether to test connections.', 'jetpack-publicize-pkg' ),
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Schema for the endpoint.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'jetpack-publicize-connection',
+			'type'       => 'object',
+			'properties' => array(
+				'connection_id'   => array(
```

Comment:
Looking at this, the endpoint exists because we need to rename a couple of fields and add some more. Would it be possible to extend [the current connection test results endpoint](https://github.com/Automattic/jetpack/blob/ceebd5264525bb78288ad3a311f3c43b0fbcbc7a/projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/publicize-connection-test-results.php), and create a version 2.1 from there?

We're copying quite a bit of code and creating a third set of endpoints, which we should probably avoid if we can.

Reply:
> Would it be possible to extend [the current connection test results endpoint](https://github.com/Automattic/jetpack/blob/ceebd5264525bb78288ad3a311f3c43b0fbcbc7a/projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/publicize-connection-test-results.php), and create a version 2.1 from there?

I initially tried to update that endpoint to enhance it but it resulted in complications for its use on all the sites - Simple and Jetpack. It also cannot be used in Standalone plugins as is.

> We're copying quite a bit of code and creating a third set of endpoints, which we should probably avoid if we can.

Yes, existing endpoints are spread all over and the logic is not consolidated, which actually resulted in the schema mess we are trying to solve. I did think about keeping the changes to minimal but it felt like creating more such mess.

The approach here tries to consolidate the endpoint which can be used everywhere - WPCOM and Jetpack sites, both in Jetpack and Social plugin.